content/en/docs/tasks/job/parallel-processing-expansion.md

```diff
@@ -177,46 +209,102 @@ spec:
+python dicts (lines 1-4).  Then a for loop emits one Job manifest for each
+set of parameters (remaining lines).
+
+This example relies on a feature of YAML. One YAML file can contain multiple
+documents (Kubernetes manifests, in this case), separated by `---` on a line
+by itself.
+You can pipe the output directly to `kubectl` to create the Job objects.
 
-You will need the jinja2 package if you do not already have it: `pip install --user jinja2`.
-Now, use this one-line python program to expand the template:
+Next, use this one-line Python program to expand the template:
 
 ```shell
 alias render_template='python -c "from jinja2 import Template; import sys; print(Template(sys.stdin.read()).render());"'
 ```
 
-
-
-The output can be saved to a file, like this:
+Use `render_template` to convert the parameters and template into a single
+YAML file containing Kubernetes manifests:
 
 ```shell
+# This requires the alias you defined earlier
 cat job.yaml.jinja2 | render_template > jobs.yaml
 ```
 
-Or sent directly to kubectl, like this:
+You can view `jobs.yaml` to verify that `render_template` is working OK.
+
+Once you are happy that `render_template` worked how you intended,
+you can pipe its output into `kubectl`:
 
 ```shell
 cat job.yaml.jinja2 | render_template | kubectl apply -f -
 ```
 
+Kubernetes accepts and runs the Jobs you created.
+
+## Clean up {#cleanup-2}
+
+```shell
+# Remove the Jobs you created
+# Your cluster automatically cleans up their Pods
+kubectl delete job -l jobgroup=jobexample
+```
+
+{{% /capture %}}
+{{% capture discussion %}}
+
+## Using Jobs in real workloads
+
+In a real use case, each Job performs some substantial computation, such as rendering a frame
+of a movie, or processing a range of rows in a database. If you were rendering a movie
+you would set `$ITEM` to the frame number. If you were processing rows from a database
+table, you would set `$ITEM` to represent the range of database rows to process.
+
+In the task, you ran a command to collect the output from Pods by fetching
+their logs. In a real use case, each Pod for a Job writes its output to
+durable storage before completing. You can use a PersistentVolume for each Job,
+or an external storage service. For example, if you are rendering frames for a movie,
+use HTTP to `PUT` the rendered frame data to a URL, using a different URL for each
+frame.
+
+## Labels on Jobs and Pods
+
+After you create a Job, Kubernetes automatically adds additional
+{{< glossary_tooltip text="labels" term_id="label" >}} that
+distinguish one Job's pods from another Job's pods.
+
+In this example, each Job and its Pod template have a label:
+`jobgroup=jobexample`.
+
+Kubernetes itself pays no attention to labels named `jobgroup`. Setting a label
+for all the Jobs you create from a template makes it convenient to operate on all
+those Jobs at once.
+You can see from the job template that each Pod also gets the same label, so that you
```

Comment:
nit: line 283 `job template` refers to the original `job-tmpl.yaml`?

Reply:
Revised